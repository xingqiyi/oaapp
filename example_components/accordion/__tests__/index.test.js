import React from 'react';
import { render, mount } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import Accordion from '../index';

describe('Accordion', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <Accordion>
        <Accordion.Panel header="标题一">
          内容一
        </Accordion.Panel>
        <Accordion.Panel header="标题二">
          内容二
        </Accordion.Panel>
      </Accordion>,
    );
    expect(renderToJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.am-accordion')).toHaveLength(1);
  });

  it('renders accordion prop correctly', () => {
    const wrapper = mount(
      <Accordion accordion>
        <Accordion.Panel header="标题一">
          内容一
        </Accordion.Panel>
        <Accordion.Panel header="标题二">
          内容二
        </Accordion.Panel>
      </Accordion>,
    );
    // accordion props make only one active panel
    wrapper.find('.am-accordion-header').at(0).simulate('click');
    wrapper.find('.am-accordion-header').at(1).simulate('click');
    expect(wrapper.find('.am-accordion-item').at(1).hasClass('am-accordion-item-active')).toBe(true);
  });
});
