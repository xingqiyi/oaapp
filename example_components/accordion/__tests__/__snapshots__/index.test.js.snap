// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion renders correctly 1`] = `
<div
  class="am-accordion"
>
  <div
    class="am-accordion-item"
  >
    <div
      aria-expanded="false"
      class="am-accordion-header"
      role="tab"
    >
      <i
        class="arrow"
      />
      标题一
    </div>
  </div>
  <div
    class="am-accordion-item"
  >
    <div
      aria-expanded="false"
      class="am-accordion-header"
      role="tab"
    >
      <i
        class="arrow"
      />
      标题二
    </div>
  </div>
</div>
`;
